solver/src/settlement/settlement_encoder.rs

```diff
@@ -373,4 +421,67 @@ pub mod tests {
         });
         assert!(encoder.add_token_equivalency(token_a, token_b).is_err());
     }
+
+    fn token(number: u64) -> H160 {
+        H160::from_low_u64_be(number)
+    }
+
+    #[test]
+    fn merge_ok() {
+        let prices = hashmap! { token(1) => 1.into(), token(3) => 3.into() };
+        let mut encoder0 = SettlementEncoder::new(prices);
+        let mut order13 = OrderBuilder::default()
+            .with_sell_token(token(1))
+            .with_buy_token(token(3))
+            .build();
+        order13.order_meta_data.uid.0[0] = 0;
+        encoder0.add_trade(order13, 13.into()).unwrap();
+        encoder0.append_to_execution_plan(NoopInteraction {});
```

Comment:
Can we also add an unwrap here for completeness?

Reply:
The function doesn't return a result so there is nothing to unwrap.

Reply:
I meant an `UnwrapInteraction`. Can you add `encoder0.add_unwrap(...)` to the testcase? Otherwise, I think we could "forget" to push the unwraps in the `merge` implementations and the unit test would still pass.

Reply:
Will do in another PR since I had auto merge enabled here.